### ISSUE TITLE
Add buttons are available for user with no rights

### ISSUE THREAD

**Issue by casmoir** [apache/superset#30927]
### Bug description

I've created custom role with no rights
![image](https://github.com/user-attachments/assets/fff81bef-93c2-4d56-8633-3e44e815efa3)

After log in user with this role is taken to the page superset/welcome/ where he can see buttons "+ DASHBOARD" and "+ CHART"
![image](https://github.com/user-attachments/assets/32e5d6f3-5849-49d9-bf42-3184b36f5ba1)

After clicking on these buttons user gets notification "Access is Denied"

Expectation: user can see buttons "+ DASHBOARD" and "+ CHART" only if he has right to create Dashboard/Chart

### Superset version

master / latest-dev

### Python version

3.9

### Node version

16

### Browser

Chrome

### Checklist

- [X] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [X] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [X] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by casmoir** [comment]
@dosu thanks for your answer, but it doesn't solve my issue

**Comment by anantmulchandani** [comment]
hey @casmoir .
I see 500 internal server error if the user has no roles. 
What superset version are you using and the what is your auth method?

**Comment by rusackas** [comment]
Agreed, it'd be far better to hide irrelevant buttons. Certainly room for improvement here, but this has been inactive for ~5 months, and I'd consider it an improvement request rather than a bug. Anyone is welcome to open PRs tackling this if they're so inclined, but for now I'll consider this not-planned. Happy to reopen it and move it to an "Ideas" thread if someone wants to discuss/recruit around it.
